plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOracleConnectorTest.java

```diff
@@ -83,4 +88,16 @@ protected SqlExecutor onRemoteDatabase()
     {
         return oracleServer::execute;
     }
+
+    @Test
+    public void testPushdownJoinWithLongNameSucceeds()
```

Comment:
Can we have a test with multiple super long column names used in the same Trino query that are almost the same (`zzz..zza`, `zzz..zzb`, ...)? Such columns should represent different data. The thing I was thinking we could check is to validate that synthetic names do not generate conflicts.

Or is it an overkill?

Reply:
yes, see the tests being added in https://github.com/trinodb/trino/pull/18984

fortunately the remote databases which silently truncate long identifiers (SQLServer, Postgres, Redshift) all prevent this bug from happening because they realise that the truncated alias is now ambiguous.